src/main/scala/kanaloa/reactive/dispatcher/Dispatcher.scala

```diff
@@ -24,13 +25,22 @@ trait Dispatcher extends Actor {
 
   protected lazy val queue = context.actorOf(queueProps, name + "-backing-queue")
 
-  private[dispatcher] val processor = context.actorOf(QueueProcessor.withCircuitBreaker(
-    queue,
-    backend,
-    settings.workerPool,
-    settings.circuitBreaker,
-    metricsCollector
-  )(resultChecker), name + "-queue-processor")
+  private[dispatcher] val processor = {
+    val props = settings.circuitBreaker.map(QueueProcessor.withCircuitBreaker(
+      queue,
+      backend,
+      settings.workerPool,
+      _,
+      metricsCollector
+    )(resultChecker)).getOrElse(QueueProcessor.default(
+      queue,
+      backend,
+      settings.workerPool,
+      metricsCollector
+    )(resultChecker))
```

Comment:
I feel like a `match` might be a bit clearer here, to make it explicit that we're handling the cases where it's present vs absent

```scala
val props = settings.circuitBreaker match {
  case None =>
    QueueProcessor.default(...)(resultChecker)
  case Some(cb) =>
    QueueProcessor.withCircuitBreaker(..., cb, ...)(resultChecker)
}
```